metric/kamailio5/kamailio.cfg

```diff
@@ -410,41 +420,64 @@ route[STORE] {
+
+        if($sht(cc=>packet::count) == $null) $sht(cc=>packet::count) = 0;
+        $sht(cc=>packet::count) = $sht(cc=>packet::count) + 1;
+
+	mq_add("queue_sip_insert", "$var(table)", "$var(homer_query_values)");
 }
 
 route[SIP_INSERT] {
-        xlog("L_CRIT","Enter queue_sip_insert\n");
+	$var(current_qsize) = mq_size("queue_sip_insert");
+	if ($var(current_qsize) == -1) {
+		$var(total_qsize) = 1;
+		xlog("L_DBG","Insert count: $sht(cc=>packet::count) \n");
+		$sht(cc=>packet::count) = 0;
+	}
+	$var(total_qsize) = $var(total_qsize) + $var(current_qsize);
+        xlog("L_DBG","Enter queue_sip_insert! Current queue size: $var(current_qsize), total queue count: $var(total_qsize) \n");
         
 	$var(sql_query_statement) = "";
 
-        while (mq_fetch("queue_sip_insert")) {
+	$var(loop_count) = 1;
+        while (mq_fetch("queue_sip_insert") && $var(loop_count) < MAX_WHILE_LOOPS) {
 		$var(table) = $mqk(queue_sip_insert);
- 		if (($(var(sql_query_statement){s.len}) + $(mqv(queue_sip_insert){s.len})) < 65000) {
+		if (!is_avp_set("$avp($var(table))")) {
+                	xlog("L_DBG","Setting table <$var(table)>\n");
+			$avp($var(table)) = "";
+			$avp($pp-tables) = $var(table);
+		}
+		$var(sql_query_length) = $(avp($var(table)){s.len}) + $(mqv(queue_sip_insert){s.len});
+ 		if ($var(sql_query_length) < (MAX_BUFFER_LENGTH-150)) {
 
                 	xlog("L_DBG","Got one Queue Key in while loop <$mqk(queue_sip_insert)>\n");
                 	xlog("L_DBG","Got one Queue Value in while loop <$mqv(queue_sip_insert)>\n");
 
-                	$var(sql_query_statement) = $var(sql_query_statement) + $mqv(queue_sip_insert) + ",";
+                	$(avp($var(table))[0]) = $(avp($var(table))[0]) + $mqv(queue_sip_insert) + ",";
 		} else {
+			$var(values) = $(avp($var(table))[0]);
 			route(DB_INSERT);
-        		$var(sql_query_statement) = "";
+                	$(avp($var(table))[0]) = $mqv(queue_sip_insert) + ",";
 		}
+		$var(loop_count) = $var(loop_count) + 1;
 		
         }
 
-	if ($var(sql_query_statement) == "") {
-		return 0;
+	while ($avp($pp-tables) != $null) {	
+		$var(table) = $avp($pp-tables);
+		$var(values) = $(avp($var(table))[0]);
+		xlog("L_DBG","Flushing table $var(table)\n");
+		route(DB_INSERT);
+		$avp($pp-tables) = $null;
 	}
-
-	route(DB_INSERT);
 }
 
 route[DB_INSERT] { 
-	$var(sql_query_statement) = $(var(sql_query_statement){s.striptail,1});
-        $var(homer_query) = "INSERT INTO " + $var(table) + " (gid, create_date, hep_header, ip_header, payload, raw) VALUES" + $var(sql_query_statement);
-        xlog("L_CRIT","homer_query length: $(var(homer_query){s.len})\n");
+	$var(values) = $(var(values){s.striptail,1});
+        $var(homer_query) = "INSERT INTO " + $var(table) + " (gid, create_date, hep_header, ip_header, payload, raw) VALUES" + $var(values);
+        xlog("L_DBG","homer_query length: $(var(homer_query){s.len})\n");
 
         sql_query("sqlhomer", "$var(homer_query)");
+
```

Comment:
We should rename $var(homer_query) to 
$var(sip_db_insert_statement) or something similar.